src/Pages/Dashboard/Sensor.jsx

```diff
@@ -1,62 +1,118 @@
+        this.setState({selected: value});
+        let offset;
+        if (value === 1) {
+            offset = 60 * 60 * 1000;
+        } else if (value === 2) {
+            offset = 2 * 60 * 60 * 1000;
+        } else if (value === 3) {
+            offset = 6 * 60 * 60 * 1000;
+        } else if (value === 4) {
+            offset = 24 * 60 * 60 * 1000;
+        } else if (value === 5) {
+            offset = 7 * 24 * 60 * 60 * 1000;
+        } else if (value === 6) {
+            offset = 30 * 24 * 60 * 60 * 1000;
+        } else if (value === 7) {
+            offset = 183 * 24 * 60 * 60 * 1000;
+        } else if (value === 8) {
+            offset = 365 * 24 * 60 * 60 * 1000;
+        } else {
+            offset = 6 * 60 * 60 * 1000;
+        }
+        let date = new Date();
+        let now = date.getTime();
+        let d = now - offset;
+        this.timestamp = Math.floor(d / 1000);
+        this.nowstamp = Math.floor(now / 1000);
+        this.setState({range: this.timestamp});
+        this.getDatabase(this.props.sensorId);
     };
-  }
-
-  // We wait until the component has been mounted (which means
-  // other components are unmounted) before polling, so we don't
-  // have multiple registers to a particular database
-  componentDidMount = () => {
-    this.getDatabase(this.props.sensorId);
-  };
-
-  getDatabase = sensorId => {
-    this.db = this.props.firebase.database().ref(sensorId);
-    let temp = [];
-    // Use .once() to make it call less data
-    this.db.limitToLast(30).once("value", e => {
-      for (let i in e.val()) {
-        temp.push(e.val()[i]);
-      }
-    });
-    this.db.limitToLast(1).on("child_added", e => {
-      temp = temp.slice();
-      if (temp.length >= 30) temp.shift();
-      temp.push(e.val());
-      this.setState({
-        data: temp
-      });
-    });
-  };
-
-  componentWillUnmount = () => {
-    this.db.off();
-  };
-
-  render() {
-    const { classes } = this.props;
-    return (
-      <div className={classes.root}>
-        <Dashboard
-          data={this.state.data}
-          sensorName={this.props.sensorName}
-          sensorId={this.props.sensorId}
-          firebase={this.props.firebase}
-        />
-      </div>
-    );
-  }
+
+    getDatabase = sensorId => {
+
+        this.db = this.props.firebase.database().ref(sensorId);
+        let temp = [];
+        let temphistory = [];
+        // Use .once() to make it call less data
+        // this.db.limitToLast(30).once("value", e => {
+        //   for (let i in e.val()) {
+        //     temp.push(e.val()[i]);
+        //   }
+        // })
+
+        this.db.orderByKey()
+            .startAt(this.timestamp.toString())
+            .endAt(this.nowstamp.toString())
+            .once("value", e => {
+                for (let i in e.val()) {
+                    temphistory.push(e.val()[i]);
+                    // temp.push(e.val()[i]); //removed so that main data page shows only live data
```

Comment:
Does this follow the behaviour you described? I haven't pulled and tested it but my first glance would make me assume that only our history tab will show the last 30 days of data, while our data tab will only show data as they come through.

Reply:
You're right I mixed up my plans versus what the code actually does. Do you want that to happen though? Or a shorter time range?